src/modules/bluefish/producer/bluefish_producer.cpp

```diff
@@ -348,7 +348,7 @@ struct bluefish_producer : boost::noncopyable
             blue_->render_buffer_capture(BlueBuffer_Image_HANC(schedule_capture_frame_id_));
             dma_ready_captured_frame_id_ = capturing_frame_id_;
             capturing_frame_id_          = schedule_capture_frame_id_;
-            schedule_capture_frame_id_   = (++schedule_capture_frame_id_ % 4);
+            schedule_capture_frame_id_   = (schedule_capture_frame_id_ + 1) % 4;

```

Comment:
schedule_capture_frame_id_ is no longer being incremented. Was this intentional?

Reply:
schedule_capture_frame_id_  was previously increased, and then overwritten by a new value. The modification of a variable is unsequenced relative to another operation on the same variable. This may lead to undefined behavior.